src/neuroglancer/viewer.ts

```diff
@@ -614,26 +615,80 @@ export class Viewer extends RefCounted implements ViewerState {
       });
     }
 
-    this.bindAction('two-point-merge', () => {
-      this.mouseState.toggleAction();
-      if (this.mouseState.actionState === ActionState.INACTIVE) {
-        this.mouseState.setMode(ActionMode.NONE);
-        StatusMessage.showTemporaryMessage('Merge mode deactivated.');
+    const isTimeDisplaced = (e: ManagedUserLayerWithSpecification) => {
+      let displayState = (<SegmentationUserLayer>e.layer).displayState;
+      return displayState && displayState.timestamp.value !== '';
+    };
+
+    const timeLock = (e: ManagedUserLayerWithSpecification, lock: boolean) => {
+      let displayState = (<SegmentationUserLayer>e.layer).displayState;
```

Comment:
If you are certain that a property exists (or handle missing properties yourself, like you do here), you can just use `e.layer.displayState!` (note the exclamation mark).
I would prefer this here over adding `SegmentationUserLayer` to the imports

Reply:
still getting prop does not exist on type errors

Reply:
The type import is necessary since, only the displaystate for segmentation user layer with graph has the timestamp info. User Layer makes TS assume this is the displaystate for the segmentation user layer